pandas/tests/tools/test_to_datetime.py

```diff
@@ -2530,3 +2530,14 @@ def test_empty_string_datetime_coerce__unit():
     # verify that no exception is raised even when errors='raise' is set
     result = to_datetime([1, ""], unit="s", errors="raise")
     tm.assert_index_equal(expected, result)
+
+
+def test_to_datetime_monotonic_increasing_index():
+    # GH28238
+    times = date_range(datetime.now(), periods=1000, freq="h")
+    times = times.to_frame(index=False, name="DT").sample(1000)
+    times.index = times.index.to_series().astype(float) / 1000
+    converted = to_datetime(times.iloc[:, 0])
+    expected = np.ravel(times.values)
+    result = np.ravel(converted.values)
+    tm.assert_equal(result, expected)
```

Comment:
Instead can you use `tm.assert_series_equal`?

Reply:
Updated